src/compiler/checker.ts

```diff
@@ -1564,7 +1566,7 @@ namespace ts {
                     error(errorNode, diag, moduleReference, resolvedModule.resolvedFileName);
                     return undefined;
                 }
-                else if (compilerOptions.noImplicitAny && moduleNotFoundError) {
+                else if (noImplicitAny && moduleNotFoundError) {

```

Comment:
This sort of thing is a bug waiting to happen - it is very likely that next time someone will need to check a flag like `noImplicitAny`, they will go through `compilerOptions` first. That said, I'm not sure what else we could do here.